syntax = "proto3";

package measurement.v1;

import "google/protobuf/timestamp.proto";

option go_package = "measurement/v1";

message Meta {
  google.protobuf.Timestamp time = 1;
  string measurement_id = 2;
}

message Climate {
  Meta meta = 1;

  float humidity = 2;
  float pressure = 3;
  float temperature_humidity = 4;
  float temperature_pressure = 5;
}

message System {
  message LA {
    float min1 = 1;
    float min5 = 2;
    float min15 = 3;
  }

  message Memory {
    int64 total_kb = 1;
    int64 used_kb = 2;
    int64 free_kb = 3;
    int64 shared_kb = 4;
    int64 cache_kb = 5;
    int64 available_kb = 6;
  }

  message Disk {
    int64 total_kb = 1;
    int64 used_kb = 2;
    int64 available_kb = 3;
    int32 use_prct = 4;
  }

  Meta meta = 1;

  float cpu_temperature = 2;
  LA la = 3;
  Memory memory = 4;
  Disk disk = 5;
}
